### PR TITLE
docs: update en astro-components about code comments

### PR DESCRIPTION
**Pull request by edv-ns**
#### What kind of changes does this PR include?

- New or updated content
- Changes to the docs site code

#### Description

- What does this PR change?

  - Update `Comments` section in astro-components page.
  - Showcase different commenting types in Astro files JSX part.
  - Add caution of which comments are included into browser DOM and which is not.

#### Preview 
  - https://deploy-preview-2229--astro-docs-2.netlify.app/en/core-concepts/astro-components/#comments

### CODE REVIEW

**Comment by edv-ns** [comment]
@delucis as discussed in [Discord](https://discord.com/channels/830184174198718474/853350631389265940/1054350850136481815) 🙏 Please review the changes.

File: src/pages/en/core-concepts/astro-components.md [hunk 1]
```diff
@@ -275,10 +275,15 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
 
 In Astro, you can use standard HTML comments where JSX would use JavaScript style comments.
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:276]:
```suggestion
In Astro, you can use standard HTML comments or JavaScript style comments.
```
I think this sentance can change to match what we're saying below.

File: src/pages/en/core-concepts/astro-components.md [hunk 2]
```diff
@@ -275,10 +275,15 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
...
 
-```html title="example.astro"
+```astro title="example.astro"
 <!-- HTML comment syntax is valid in .astro files -->
+{/* JS comment  */}
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:280]:
```suggestion
{/* JS comment syntax is also valid */}
```

File: src/pages/en/core-concepts/astro-components.md [hunk 3]
```diff
@@ -275,10 +275,15 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
 
 In Astro, you can use standard HTML comments where JSX would use JavaScript style comments.
 
-```html title="example.astro"
+```astro title="example.astro"
 <!-- HTML comment syntax is valid in .astro files -->
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:279]:
```suggestion
---
---
<!-- HTML comment syntax is valid in .astro files -->
```
It might be good to show that these comments are in the templating section of an Astro file.

File: src/pages/en/core-concepts/astro-components.md [hunk 4]
```diff
@@ -275,10 +275,15 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
...
+{/* JS comment  */}
 ```
 
+:::caution
+The HTML comments will be included in browser DOM, while JS ones will be skipped. In case you need to leave TODO messages or other development-only explanations in JSX part, use JavaScript style comments instead.
+:::
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:285]:
```suggestion
:::caution
HTML-style comments will be included in browser DOM, while JS ones will be skipped. To leave TODO messages or other development-only explanations, you may wish to use JavaScript-style comments instead.
:::
```

**Comment by sarah11918** [comment]
Thanks, @edv-ns! This will surely be helpful to people (and, maybe prevent some embarrasing notes shipping to the client)!

I've made some editing suggestions. See what you think!

File: src/pages/en/core-concepts/astro-components.md (hunk 4 above)
Comment (edv-ns) [reply src/pages/en/core-concepts/astro-components.md:285]: Thanks for much better copy. Writing texta is not my strongest part :) will push changes accordingly soon! 👌

Comment (edv-ns) [reply src/pages/en/core-concepts/astro-components.md:285]: Actually there is one edge case when even js comments could end up in the html, which is [is:raw](https://docs.astro.build/en/reference/directives-reference/#israw) directive, but I am pretty sure it is self-explanatory on it's own, so did not include in this docs section.

File: src/pages/en/core-concepts/astro-components.md [hunk 5]
```diff
@@ -273,12 +273,19 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
 
 #### Comments
 
-In Astro, you can use standard HTML comments where JSX would use JavaScript style comments.
+In Astro, you can use standard HTML comments or JavaScript style comments.
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:276]:
```suggestion
In Astro, you can use standard HTML comments or JavaScript-style comments.
```
Just keeping our hyphens consistent! 💪

**Comment by sarah11918** [comment]
Looks great, @edv-ns! Thank you for this, and welcome to Team Docs! 🥳

File: src/pages/en/core-concepts/astro-components.md (hunk 4 above)
Comment (sarah11918) [reply src/pages/en/core-concepts/astro-components.md]: Alright, let me take one more pass at it then, without making it too wordy....  😄

File: src/pages/en/core-concepts/astro-components.md [hunk 6]
```diff
@@ -273,12 +273,19 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
...
-```html title="example.astro"
+```astro title="example.astro"
+---
+---
 <!-- HTML comment syntax is valid in .astro files -->
+{/* JS comment syntax is also valid */}
 ```
 
+:::caution
+HTML-style comments will be included in browser DOM, while JS ones will be skipped. To leave TODO messages or other development-only explanations, you may wish to use JavaScript-style comments instead.
```
Comment (sarah11918) [inline src/pages/en/core-concepts/astro-components.md:286]:
```suggestion
HTML-style comments will be included in the browser DOM when compiled, while JS ones will be skipped. To leave TODO messages or other development-only explanations, you may wish to use JavaScript-style comments instead.
```
Would you say this phrase is correct, to describe non-raw HTML only? I'm trying to think of the super-shortest thing we can say that's both true and helpful!

Comment (edv-ns) [reply src/pages/en/core-concepts/astro-components.md:286]: Yes, your version is better one! We may commit it.

Comment (edv-ns) [reply src/pages/en/core-concepts/astro-components.md:286]: Would you mind merging that in too?

ah, I see we have merged this into main 🙂
